src/aclk/mqtt_websockets/ws_client.c

```diff
@@ -158,48 +161,27 @@ int ws_client_want_write(ws_client *client)
     return rbuf_bytes_available(client->buf_write);
 }
 
-#define RAND_SRC "/dev/urandom"
-static int ws_client_get_nonce(ws_client *client, char *dest, unsigned int size)
-{
-    // we do not need crypto secure random here
-    // it's just used for protocol negotiation
-    int rd;
-    int f = open(RAND_SRC, O_RDONLY | O_CLOEXEC);
-    if (f < 0) {
-        ERROR("Error opening \"%s\". Err: \"%s\"", RAND_SRC, strerror(errno));
-        return -2;
-    }
-
-    if ((rd = read(f, dest, size)) > 0) {
-        close(f);
-        return rd;
-    }
-    close(f);
-    return -1;
-}
-
 #define WEBSOCKET_NONCE_SIZE 16
 #define TEMP_BUF_SIZE 4096
 int ws_client_start_handshake(ws_client *client)
 {
-    char nonce[WEBSOCKET_NONCE_SIZE];
+    nd_uuid_t nonce;
     char nonce_b64[256];
     char second[TEMP_BUF_SIZE];
     unsigned int md_len;
     unsigned char *digest;
     EVP_MD_CTX *md_ctx;
     const EVP_MD *md;
 
-    if(!*client->host) {
+    if(!client->host || !*client->host) {
         ERROR("Hostname has not been set. We should not be able to come here!");
         return 1;
     }
 
-    ws_client_get_nonce(client, nonce, WEBSOCKET_NONCE_SIZE);
+    uuid_generate_random(nonce);
     EVP_EncodeBlock((unsigned char *)nonce_b64, (const unsigned char *)nonce, WEBSOCKET_NONCE_SIZE);
-    snprintf(second, TEMP_BUF_SIZE, websocket_upgrage_hdr,
-        *client->host,
-        nonce_b64);
+    snprintf(second, TEMP_BUF_SIZE, websocket_upgrage_hdr, *client->host, nonce_b64);
```

Comment:
Two questions here:
- Why are we using `snprintf` instead our `snprintfz`?
- Considering that the majority of our code is using for size `size - 1`, should we do the same here?

Reply:
I have another PR that further improves this (eg. removing pthread_locks) so I will add these changes to that PR